designs/PackageVulnerability/FlagVulnerablePackages.md

```diff
@@ -0,0 +1,142 @@
+  * **Surface package vulnerability information on-demand**:
+    * as part of package metadata in `v3` registration blobs (`JSON`)
+    * by showing vulnerability information to package owners (only!) on package details page and version history
+  * Basic (e-mail) notification to notify package publishers for any new vulnerability detected in one or more of their packages
+
+#### ❌ Out of scope (Stage 1)
+
+Some enhancements to this feature are out of scope for the initial implementation of this feature because there are several unknowns still to be clarified. See also the *Open Questions* section below.
+
+* Enable `restore` to Inform or Warn (see Stage 2)
+* Expose vulnerability metadata in nuget.org *Search* service (not required for MVP)
+* Display vulnerability information for *transitive packages* in PMUI (additional UX complexity)
+* Expose vulnerability information *publicly* on nuget.org's package details page and version history (disclosure policies TBD)
+
+### Stage 2
+
+Stage 2 of this feature will focus on *highlighting known vulnerabilities when the package is already installed*. This involves introducing new validations at package `restore` time, and new ways of dealing with them.
+
+#### ✔️ In scope (Stage 2)
+
+* For CLI:
+  * Output warnings or errors when running `dotnet add package` (which actually does a *restore*)
+
+* For Visual Studio:
+  * Inform consumers about known vulnerabilities at time of installation or restore:
+    * Warn (⚠️) for `critical` and `high` impact vulnerabilities (minimal info + command/way to uncover detailed information)
+    * Info (ℹ️) for lower impact vulnerabilities (minimal info + command/way to uncover detailed information)
+  * Display vulnerability information for *transitive packages* in PMUI
+  * New settings to enable/disable warnings for the various impact levels of vulnerabilities
+
+* For nuget.org:
+  * Show vulnerability information in packages to their publishers on nuget.org
+    * Show the vulnerability info on the package page and version history
+
+#### ❓ Open questions (Stage 2)
+
+* For `restore`, should it be opt-in or opt-out (preferred for security features)?
+* For `dotnet add package`, define behavior (info, warn, error?) when encountering packages known to be vulnerable.
+* Need for another command (e.g. `dotnet nuget audit`) to create an audit report for deprecated, vulnerable and may be outdated packages. And a way to fix the vulnerabilities with a single command/click.
+* Disclosure policies if and when we start consuming non-disclosed vulnerability data? (e.g. under review by package author) - GH API currently doesn't surface these
+
+| Feature | State |
+|:--- |:--- |
+| A way to specify another source for vulnerability information | Backlog |
+| `why` command to understand why a package is shown as a dependency for my project | Backlog |
+| Ability to report a vulnerability on nuget.org | Discuss |
```

Comment:
NuGet package page requires a "Report a vulnerability" feature that is triaged by someone on the NuGet team.

Otherwise we need to provide instructions for https://docs.github.com/en/github/managing-security-vulnerabilities/creating-a-security-advisory

Reply:
I think the second option is more robust as it gives a single point of truth. In a scenario where we act as a moderator, there's more opportunity for lag before a new vulnerability is surfaced in our clients.